freud/environment.pyx

```diff
@@ -532,11 +532,56 @@ cdef class EnvironmentCluster(_MatchEnv):
                 global_search=False):
         r"""Determine clusters of particles with matching environments.
 
-        In general, it is recommended to specify a number of neighbors rather
-        than just a distance cutoff as part of your neighbor querying when
-        performing this computation. Using a distance cutoff alone could easily
-        lead to situations where a point doesn't match a cluster because a
-        required neighbor is just outside the cutoff.
+        An environment is defined by the bond vectors between a particle and its
+        neighbors as defined by :code:`env_neighbors`.
+        For example, :code:`env_neighbors= {'num_neighbors': 8}` means that every
+        particle's local environment is defined by its 8 nearest neighbors.
+        Then, each particle's environment is compared to the environments of a
+        selected list of other particles in the system, controlled by the
+        :code:`neighbors` parameter. For example, :code:`neighbors={'r_max': 3.0}`
+        means that the environment of each particle will be compared to the
+        environment of every particle within a distance of 3.0.
+
+        Two environments are compared using `point-set registration
+        <https://en.wikipedia.org/wiki/Point-set_registration)>`_.
+        Specifically, we use the `Kabsch algorithm
+        <https://en.wikipedia.org/wiki/Kabsch_algorithm>`_ to find the optimal
+        rotation matrix mapping one environment onto another. The Kabsch
+        algorithm assumes a one-to-one correspondence between points in the two
+        environments. However, we typically do not know which point in one environment
+        corresponds to a particular point in the other environment. The brute force
+        solution is to try all possible correspondences, but the resulting
+        combinatorial explosion makes this problem infeasible, so we use a tolerance
+        :code:`threshold` to terminate the search when we have found a permutation
+        of points that results in a sufficiently low RMSD.
+
+        .. note::
+
+            Using a distance cutoff for :code:`env_neighbors` could
+            lead to situations where the :code:`cluster_environments`
+            contain different numbers of particles. In this case, the
```

Comment:
Instead of "Using a distance cutoff for :code:`env_neighbors` could lead to situations where the :code:`cluster_environments` contain different numbers of particles", I would say:

"Using a distance cutoff for :code:`env_neighbors` could lead to situations where the :code:`cluster_environments` contain different numbers of vectors."

Or instead of vectors above, you could write "neighbors." Either choice removes ambiguity about whether we are referring to the number of particles in each cluster, or the number of particles that define environments.